src/flows/pipes/Visualization/threshold.ts

```diff
@@ -4,85 +4,141 @@ export enum ThresholdFormat {
   Deadman = 'deadman',
 }
 
+export enum ThresholdType {
```

Comment:
not agreeing with it existing in the visualization panel, but why isnt this using the same source of truth for types as is generating the thresholds? `keyof COMMON_THRESHOLD` or whatever
https://github.com/influxdata/ui/blob/master/src/flows/pipes/Visualization/threshold.ts#L21

Reply:
not sure what you mean.

I would need to keyof another type definition (interface definition).
https://www.typescriptlang.org/docs/handbook/2/keyof-types.html

Reply:
you added a lot of changes to the threshold definition object to tie it to the enum so that you could later use the enum values to validate a collective type. seems like a lot of effort to now couple these two concepts (the type definitions and the thing that validates to a key in that object) through a new intermediary type definition. how you have this defined, you also have to update the enum anytime you want to modify the object, which feels like an inversion of responsibility. it'd be quicker, easier to maintain, and involve less coupling to use the original object that serves to define the threshold types as the source of truth and utilize the typescript functionality of `keyof` to generate the type definition that this enum fills, but automatically. the definition then changes from:
```
export enum ThresholdType = { ... }
```

to:
```
export type ThresholdType = keyof typeof COMMON_THRESHOLD_TYPES
```

i was wondering if there was a use case i missed that would justify using the enum vs the keyof